### ISSUE TITLE
Implement incremental xDS in Envoy

### ISSUE THREAD

**Issue by fredlas** [envoyproxy/envoy#4991]
The incremental xDS design has been decided on, and added to the appropriate .proto file (in #3470). Now, incremental xDS clients should be implemented in Envoy. I am currently getting started on CDS.

*Relevant Links*: an early draft proposal for incremental xDS, still relevant for examples and overview of goals: https://docs.google.com/document/d/1dh8ACRzNNUkVYB3exepTPlLQK6CyDrOaP2OwwGlxDB4/edit#heading=h.yfahbzhf2783

**Comment by fredlas** [comment]
ongoing

**Comment by hzxuzhonghu** [comment]
Looking forward to this.

**Comment by fredlas** [comment]
A list of subtasks to be done in the course of "fully implement incremental xDS", starting with what I am currently working on:

- [ ] Merge a working implementation of incremental CDS in Envoy (#5466)
- [ ] Convert all standard xDS implementations to use the incremental-style onConfigUpdate callback (translating by saying "everything was removed and these things were added"), to remove the hack of having two types of onConfigUpdate
- [ ] Change the request/response protobufs to support some form of partial resource rejection (#5739)
- [ ] Implement incremental versions of all other xDSes
- [ ] Implement ADS, and presumably rewrite non-aggregated incremental xDSes as "ADS that happens to only carry one xDS", like how standard xDS currently is

**Comment by mattklein123** [comment]
@fredlas @htuch what's the current status of this issue?

**Comment by fredlas** [comment]
#7293, which depends on #7108, will be marking this issue fixed. As for the nice-to-have "Unify implementations (including aggregated) of state-of-the-world and delta" checkbox item, I have some not-yet-in-PR code written up that will accomplish that cleanly and simply.
